docs/package/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import msvc_runtime_flag
+import functools
+import os
+
+required_conan_version = ">=1.36.0"
+
+class packageConan(ConanFile):
+    name = "package"
+    description = "shortd escription"
+    license = "" # conform to SPDX License List
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/project/package"
+    topics = ("topic1", "topic2") # no "conan" in topics
+    settings = "os", "arch", "compiler", "build_type" # even for header only
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake", "cmake_find_package_multi"
+
+    # no manual caching of build helper like  _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _is_msvc(self):
```

Comment:
On Conan 1.45 there is a native version: https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc

Soon, 1.45.0 will be available on Conan Center Index.

Reply:
Now it is.